### PR TITLE
fix: fix get_resources not adding usps_mail_delivered to configuration selection

### PR DESCRIPTION
**Pull request by gpeavy**
## Proposed change

Fix bug in delivered USPS mail implementation.

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests
- [ ] Documentation update
- [ ] Adds a new shipper
- [ ] Update existing shipper

## Additional information

fetch() is never called with sensor="usps_mail_delivered" in process_emails() as resources does not contain the BinarySensors.
This is probably just a temporary fix.

- This PR fixes or closes issue: fixes #
- This PR is related to issue: #915

### CODE REVIEW

File: custom_components/mail_and_packages/helpers.py [hunk 1]
```diff
@@ -217,6 +217,12 @@ def process_emails(hass: HomeAssistant, config: ConfigEntry) -> dict:
         except Exception as err:
             _LOGGER.error("Error updating sensor: %s reason: %s", sensor, err)
 
+    # Update usps_mail_delivered
+    try:
+        fetch(hass, config, account, data, "usps_mail_delivered")
```
Comment (firstof9) [inline custom_components/mail_and_packages/helpers.py:222]: The sensor would already be in the `resources` and searched for twice.

Comment (gpeavy) [reply custom_components/mail_and_packages/helpers.py:222]: I added some additional logging in my testing and I don't think it's in `resources`.  If `resources` is set by `get_resources()`, `SENSOR_TYPES` does not contain the BinarySensors. I think.

Comment (gpeavy) [reply custom_components/mail_and_packages/helpers.py:222]: Please correct me in I am wrong.  This is my first time digging into HA. Does `config.get(CONF_RESOURCES)` get set by:
https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/blob/4a397286698f3b3ea4d26542b279071b25716284/custom_components/mail_and_packages/config_flow.py#L211-L213
If so `get_resources()` here:
https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/blob/4a397286698f3b3ea4d26542b279071b25716284/custom_components/mail_and_packages/helpers.py#L94-L103
uses `SENSOR_TYPES` defined here:
https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/blob/4a397286698f3b3ea4d26542b279071b25716284/custom_components/mail_and_packages/const.py#L651
which does not include `usps_mail_delivered` defined here:
https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/blob/4a397286698f3b3ea4d26542b279071b25716284/custom_components/mail_and_packages/const.py#L1157

Comment (firstof9) [reply custom_components/mail_and_packages/helpers.py:222]: Ah you are correct, it's been excluded from the resources in error. That's where the bug needs to be fixed.

Comment (gpeavy) [reply custom_components/mail_and_packages/helpers.py:222]: That was my guess. I wasn't exactly sure how to do that.

Comment (gpeavy) [reply custom_components/mail_and_packages/helpers.py:222]: In addition to the Binary sensors, I think the Images sensors are missing too.

Comment (firstof9) [reply custom_components/mail_and_packages/helpers.py:222]: I put in a PR on your fork that would solve the issue.

Comment (firstof9) [reply custom_components/mail_and_packages/helpers.py:222]: > In addition to the Binary sensors, I think the Images sensors might be missing too, as in they are always created and not selected during setup.

This is by design (for now).

Comment (gpeavy) [reply custom_components/mail_and_packages/helpers.py:222]: > I put in a PR on your fork that would solve the issue.

I'll pull it in and give it a try.

Comment (firstof9) [reply custom_components/mail_and_packages/helpers.py:222]: Feel free to give it a run on a test HA instance if you like too, all the sensors will now be sorted alphabetically.

File: custom_components/mail_and_packages/const.py [hunk 2]
```diff
@@ -4,10 +4,8 @@
 
 from typing import Final
 
-from homeassistant.components.binary_sensor import (
-    BinarySensorDeviceClass,
-    BinarySensorEntityDescription,
-)
+from .entity import MailandPackagesBinarySensorEntityDescription
```
Comment (firstof9) [inline custom_components/mail_and_packages/const.py:7]:
```suggestion
from homeassistant.components.binary_sensor import BinarySensorDeviceClass
```

File: custom_components/mail_and_packages/const.py [hunk 3]
```diff
@@ -4,10 +4,8 @@
...
+from homeassistant.components.binary_sensor import BinarySensorDeviceClass
```
Comment (firstof9) [inline custom_components/mail_and_packages/const.py:8]:
```suggestion
from homeassistant.components.sensor import SensorDeviceClass, SensorEntityDescription
```

File: custom_components/mail_and_packages/const.py [hunk 4]
```diff
@@ -4,10 +4,8 @@
...
 from homeassistant.components.sensor import SensorDeviceClass, SensorEntityDescription
```
Comment (firstof9) [inline custom_components/mail_and_packages/const.py:9]:
```suggestion
from homeassistant.helpers.entity import EntityCategory
```

File: custom_components/mail_and_packages/const.py [hunk 5]
```diff
@@ -4,10 +4,8 @@
...
 from homeassistant.helpers.entity import EntityCategory
```
Comment (firstof9) [inline custom_components/mail_and_packages/const.py:10]:
```suggestion
```

File: custom_components/mail_and_packages/const.py [hunk 6]
```diff
@@ -4,10 +4,8 @@
...
```
Comment (firstof9) [inline custom_components/mail_and_packages/const.py:11]:
```suggestion
from .entity import MailandPackagesBinarySensorEntityDescription
```

**Review by firstof9 — CHANGES_REQUESTED** [review]
This should fix the import linting errors.

File: custom_components/mail_and_packages/helpers.py (hunk 1 above)
Comment (gpeavy) [reply custom_components/mail_and_packages/helpers.py]: I've finally started to test it now.  Running into a weird situation where the Logbook and History show the sensor turning on, but the sensor still shows off. 
image

Comment (firstof9) [reply custom_components/mail_and_packages/helpers.py]: if you refresh does it show "on"?
if so, this will be an bug in HA not updating the front end.

Comment (gpeavy) [reply custom_components/mail_and_packages/helpers.py]: Interesting.  Yep. Issue with HA front end. I had clicked around to different views in HA and even restarted the Mail and Packages integration, but it wasn't until I refreshed the browser that it updated.

Comment (firstof9) [reply custom_components/mail_and_packages/helpers.py]: I've noticed this issue after the 2024.8.2 update.

**Pull request closed without merging**